pkg/api/controllers/profile.go

```diff
@@ -54,38 +55,40 @@ func (p *ProfilePortal) CreateProfile() {
 	}
 
 	// Validate StorageType is block or file
-	if profile.StorageType != constants.Block && profile.StorageType != constants.File {
-		errMsg := fmt.Sprintf("parse profile request body failed : %v is invalid storagetype", profile.StorageType)
-		p.ErrorHandle(model.ErrorBadRequest, errMsg)
-		return
-	}
-
-	capabilityflag := true
-	if profile.StorageType == constants.File {
-		if pp := profile.ProvisioningProperties; !pp.IsEmpty() {
+	stype := strings.ToLower(profile.StorageType)
+	switch stype {
+	case constants.Block:
+		break
+	case constants.File:
+		if pp := profile.ProvisioningProperties; pp.IsEmpty() {
```

Comment:
Is this required?  We should support default ones if this is not defined. Also admin can define custom properties if pre-defined ones are not used.

Reply:
It is to check the StorageAccessCapability, if we won't follow the order then it will consider it as custom property so just to follow the order of properties specification i have added ProvisioningProperties and datastorage. Those are all not pre-defined. Admin still can add custom properties.